src/libprojplugin/libprojplugin.h

```diff
@@ -34,13 +37,19 @@ private slots:
 private:
     bool isRw; //1 - rw, 0 - ro
     bool erroneousState;
-    QVector<QFile *> instancesOfFiles;
-    QVariantMap parsedMetadata;
-    QStringList filesOfProject;
+
+    static QVariantMap parsedMetadata;
+    static QVector<QFile *> files;
+    QString * projectFilename;
```

Comment:
Whe you see QString*, std::string*, some_other_lib::String*, remember, it's strange :)